### DISCUSSION TITLE
Dictionary not found

### DISCUSSION THREAD

**Discussion by Overloader6** [nasa/openmct#4045] [Community Q&A]
I'll preface this with: Im pretty new to Javascript, node and pretty much everything. I've followed the tutorial and now am trying to do something myself. Form [here](https://github.com/nasa/openmct/issues/3159) i've gathered that i cant use the http thing from the tutorial. 

so im doing this 
``` javascript
function getDictionary() {
    console.log();
    return fetch("dictionary.json").then(function(response){
        return response.json()
    });
}
```

but i get `plugin.js?2852:3 GET http://localhost:8080/dictionary.json 404 (Not Found)`

Im sure im making a stupid newbie mistake but still id be really grateful for help

**Comment by Overloader6** [comment]
For now as a workaround I hardcoded the json but that cant be the solution

**Comment by unlikelyzero** [comment]
@Voreck could you post your code so we can try to reproduce

**Comment by Overloader6** [comment]
This all lies in opencmt/src/plugins/my-plugin/src/plugin.js (Idk if thats the right place to put it, I just did what seemed right to me

``` javascript
function getDictionary() {
    return fetch("dictionary.json").then(function(response){
        return response.json()
    });
}

const objectProvider = {
    get: function (identifier) {
        if (identifier.key === "dir.root") {
            return Promise.resolve({
                identifier: identifier,
                name: dictionary.name,
                type: "folder",
                location: "ROOT"
            });
        } 
    }
};

export default function () {
    return function install(openmct) {
        openmct.objects.addRoot({
            namespace: "vck.zll",
            key: "dir.root"
        });
        openmct.objects.addProvider("vck.zll", objectProvider);
    }
};

```

I've tried both simply dictionary.json as well as src/plugins/my-plugin/src/dictionary.json which is where the file lies

**Comment by Overloader6** [comment]
I found another, similar behaviour: I previously "installed" my plugin by mimicing what i saw in the plugins.js. But if I try to install it the way its installed in the tutorial, through the script tag and the install function I get a 404 at the script tag

**Comment by Overloader6** [comment]
Alright, I found a solution: I used the
``` javascript
define([
    "./dictionary.json"
], function (
    dictionary
) {
    // Something
});
```
form I found in other js files. Not a fix to the problem with the script tag though, obviouslt

**Comment by jvigliotta** [comment]
Yeah, not a bad solution. The reason we usually don't run into this type of thing is we use the strategy of [Open MCT as a dependency](https://github.com/deeptailor/openmct-as-a-dependency) this way when you're installing plugins that are outside of the Open MCT framework you're still inside of a JS environment (src/index.js in this case), so you can get past those issues of needing to use a source tag. We use webpack for our builds as well, which, could move your static dictionary file over to a built "dist" directory that holds all the code and just references it relatively.

**Reply by Overloader6** [reply]
Ok, so I tried making a Hello world plugin this way myself but openmct never starts. So I cloned the repository you linked and tried `npm install` and `npm start` there but even though I dont get errors, mct never starts

**Reply by Overloader6** [reply]
the last thing the console shows is `｢wdm｣: Compiled successfully.`

**Reply by Overloader6** [reply]
Ok, so if anyone else stumbles upon this problem as well: in the example it is defined to run on port 9000 instead of port 8080

**Comment by Overloader6** [comment]
Ok, so that makes sense to me if an user only wants to install one plugin. But what if the user wants to install more plugins? Can the user just put both plugins in the same directory, type `npm install`, opencmt gets downloaded since its a dependency (at least as far as I've understood how it works) and the user can use both plugins

Probably not the place to ask these questions, sorry

**Comment by jvigliotta** [comment]
With that kinda setup, you can install as many plugins as you like. You just need to import them in the index.js, like here: https://github.com/deeptailor/openmct-as-a-dependency/blob/8546acb2b632e8154f81d0dcb729d4bcbd2ef871/src/index.js#L2 
You could install multiple before the `openmct.start();` call.

No worries, I'll actually move this over to our Discussions page for informational purposes.

**Comment by Overloader6** [comment]
I think there should be some mention of openmct as a dependency in the tutorial
